TODO

```diff
@@ -132,6 +132,10 @@ Deprecations and removals:
 
 Features:
 
+* add a new ExecStart= flag that inserts the configured user's shell as first
+  word in the comand line. (maybe use character '.'). Usecase: tool such as
+  uid0 can use that to spawn the target user's default shell.
```

Comment:
Hmm, I think such concept isn't technically a flag, as it changes the executable we use. Others just modify the environment where the provided `ExecStart=` line is run.

Maybe just allow `$SHELL` to be specified as the executable? This is self-explanatory and avoid the "mixture" of different concepts

Reply:
we do not allow argv[0] to be configured as env var. it must be a proper path or filename. This restriction we currently enforce since for socket activation we must be able to read the selinux label off the binary ahead of time, when binding the socket, and we cannot do that if the binary is referenced via an env var, since the list of env vars is only prepared the moment we actually spawn a process for a multitude of sources...

hence my idea to use "." as a special flag for this. It would still mean we couldn't ahead of time determine the label for the process if this functionality is used (because we cannot do getpwnam() from PID 1 to figure out the services user's shell). But I think we could simply use the label of `/bin/sh` no matter what, under the assumption that all shells are labelled the same. Which isn't perfect but good enough I guess, and we shouldn't allow selinux to put us in a corner we can never get out of again

Reply:
> we do not allow argv[0] to be configured as env var. it must be a proper path or filename.

I'm not saying that we'll really do env expansion on this. It's just a more intuitive specifier in my eyes 🤔

Reply:
> But I think we could simply use the label of /bin/sh no matter what, under the assumption that all shells are labelled the same. Which isn't perfect but good enough I guess, and we shouldn't allow selinux to put us in a corner we can never get out of again

This sounds good.